torch/distributed/checkpoint/default_planner.py

```diff
@@ -258,7 +258,7 @@ def _should_include_key(self, key: str, metadata: Metadata) -> bool:
         for unflattened_key in planner_data:
             if unflattened_keys:
                 unflattened_keys.append(
-                    ".".join([unflattened_keys[-1], unflattened_key])
+                    ".".join([unflattened_keys[-1], str(unflattened_key)])
```

Comment:
this should probably be an f-string though:
```suggestion
                    f"{unflattened_keys[-1]}.{unflattened_key!s}"
```